### ISSUE TITLE
Method onSupportNavigateUp: Sequence of method arguments interchanged

### ISSUE THREAD

**Issue by croworc** [udacity/andfun-kotlin-android-trivia#8]
Sequence of method arguments interchanged:

    Class:  `MainActivity`  
    Method: `onSupportNavigateUp()`

Change from:
    ` return NavigationUI.navigateUp(drawerLayout, navController)`
to:
    `return NavigationUI.navigateUp(navController, drawerLayout)`

Branches affected:
==============
<pre>
Step.09-Solution-Adding-the-Navigation-Drawer
    Line 44
</pre>
<pre>
Step.10-Exercise-Using-Navigation-Listeners
    Line 48
</pre>
<pre>
Step.10-Solution-Using-Navigation-Listeners
    Line 54
</pre>
<pre>
Step.11-Exercise-Adding-Animation
    Line 54
</pre>
<pre>
Step.11-Solution-Adding-Animation
    Line 54
</pre>

**Comment by nicholeb** [comment]
Ridiculous that this isn't fixed after nearly two years.

**Comment by croworc** [comment]
@nicholeb :
I  assume you know the following, however, just in case you don't:
Google offers a series of [`codelab` based  Android with Kotlin courses](https://developer.android.com/courses/).
They started w/ Android Development Fundamentals, added Advanced Android Development and are just now in the process of adding more and more units to an Android Development Basics course (units 1 to 3 being implemented so far).
These courses are closely mirroring the Udacity courses, however, the repos are more up-to-date and the `codelabs` offer easily digestible chunks of learning material.
Also, the quality of the documentation & the presentation is really good. So, these courses can be a very good add-on for the Udacity courses (or can be consumed stand-alone, of course) and can help digest and understand better what's going on.
You might want to give these a try!
Cheers!

**Comment by nicholeb** [comment]
Yes! I did the first unit when it was released and just a couple of weeks ago saw that the next two had been published and have finished those as well. I've been surprised at how helpful they are in reinforcing the concepts - or sometimes I don't understand the concepts at all after doing a Udacity lesson but it makes complete sense in the codelab.

**Comment by Nickrader** [comment]
Also the solution is ?not opitmal?, seems more a workaround not using binding than addressing the transpose error.
_MainActivity.kt_
`appBarConfiguration = AppBarConfiguration(navController.graph, drawerLayout)`
`...`
`return NavigationUI.navigateUp(navController, appBarConfiguration)`
https://github.com/udacity/andfun-kotlin-android-trivia/tree/Step.09-Solution-Adding-the-Navigation-Drawer

**Comment by SudKul** [comment]
Looks like the proposed suggestions to change the arguments in `navigateUp()` are already present in the code across all branches:

```    override fun onSupportNavigateUp(): Boolean {
        val navController = this.findNavController(R.id.myNavHostFragment)
        return NavigationUI.navigateUp(navController, appBarConfiguration)
    }
```
